scripts/fun_index.R

```diff
@@ -51,3 +53,31 @@ create_html_note <- function(x, toggle = FALSE) {
     sep = "\n"
   )
 }
+
+# function to order levels in a dataframe column such that strings matching
+# a pattern are always the first / last ones (controlled via leading)
+# default leading FALSE as this is needed for the plots
+give_priority_in_order <- function(vec, pattern, leading = FALSE,
+  decreasing = TRUE) {
+
+  # extract all variables that look like "Others". This should probably be only
+  # one, but currently there is also lowercase "others"
+  # FIXME This could probably be done more elegantly
+  others_str <- vec[str_detect(vec, pattern)] %>%
+    as.character()
+
+  rest_str <- vec[!str_detect(vec, pattern)] %>%
```

Comment:
Maybe use `dplyr::anti_join` so you don't have to manually invert the condition, but maybe that's a little too much.
(Scheme's `partition` would be nice here, returning two complementary values.)

Reply:
Ah or just `setdiff`, that should be the thing for vectors...